### PR TITLE
binlog: fix show pump/drainer status

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/42643

Problem Summary:

### What is changed and how it works?
In the past code we directly use tidb's global config path to create etcd registry. After this PR we use binlog client's etcd registry to do the following staff. Binlog client will update etcd urls in time.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

Side effects

- None

Documentation

- None

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: executor/show.go [hunk 1]
```diff
@@ -1900,18 +1903,21 @@ func (e *ShowExec) fetchShowPumpOrDrainerStatus(kind string) error {
 	return nil
 }
 
-// createRegistry returns an ectd registry
-func createRegistry(urls string) (*node.EtcdRegistry, error) {
+// createRegistry returns an ectd registry, need to close, and error
+func createRegistry(urls string) (*node.EtcdRegistry, bool, error) {
+	if pumpClient := binloginfo.GetPumpsClient(); pumpClient != nil && pumpClient.EtcdRegistry != nil {
```
Comment (okJiang) [inline executor/show.go:1908]: Do we need tolerate `pumpClient` is nil?

File: executor/show.go [hunk 2]
```diff
@@ -1876,16 +1877,18 @@ func (e *ShowExec) fetchShowWarnings(errOnly bool) error {
 
 // fetchShowPumpOrDrainerStatus gets status of all pumps or drainers and fill them into e.rows.
 func (e *ShowExec) fetchShowPumpOrDrainerStatus(kind string) error {
-	registry, err := createRegistry(config.GetGlobalConfig().Path)
+	registry, needToClose, err := createRegistry(config.GetGlobalConfig().Path)
```
Comment (okJiang) [inline executor/show.go:1880]: Is it better to use a new function? This client, I think, is used for Pump/Drainer specifically.

File: executor/show.go (hunk 1 above)
Comment (okJiang) [reply executor/show.go:1908]: `createRegistry` is used in `excutor/change.go` too. Should it use the `PumpsClient`?

File: executor/show.go (hunk 2 above)
Comment (lichunzhu) [reply executor/show.go:1880]: How about changing function's name to avoid abuse?

**Review by okJiang — APPROVED** [review]

**Pull request merged**